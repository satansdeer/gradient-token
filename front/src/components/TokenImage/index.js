import React from "react";
import PropTypes from "prop-types";

const IMAGE_SIZE = 100;

const TokenImage = ({ outer, inner, size }) => {
  const gradId = `${outer}${inner}`.replace(/#/g, "");
  const imageSize = size || IMAGE_SIZE;
  return (
    <svg
      width={imageSize}
      height={imageSize}
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <radialGradient id={gradId}>
          <stop offset="0%" stopColor={outer} />
          <stop offset="100%" stopColor={inner} />
        </radialGradient>
      </defs>
      <circle
        cx={imageSize / 2}
        cy={imageSize / 2}
        r={imageSize / 2 - 4}
        fill={`url(#${gradId})`}
        stroke="#fff"
        strokeWidth="2"
      />
    </svg>
  );
};

TokenImage.propTypes = {
  outer: PropTypes.string.isRequired,
  inner: PropTypes.string.isRequired,
  size: PropTypes.number
};

export default TokenImage;
