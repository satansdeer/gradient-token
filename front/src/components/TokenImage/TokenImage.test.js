import React from "react";
import renderer from "react-test-renderer";
import TokenImage from "./index";

describe("<TokenImage />", () => {
  it("renders and matches the snapshot", () => {
    const component = renderer.create(<TokenImage outer="#fff" inner="#000" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("adjusts size based on passed size prop", () => {
    const component = renderer.create(
      <TokenImage outer="#fff" inner="#000" size={200} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
