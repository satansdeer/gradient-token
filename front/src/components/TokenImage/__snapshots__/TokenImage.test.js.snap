// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenImage /> adjusts size based on passed size prop 1`] = `
<svg
  height={200}
  version="1.1"
  width={200}
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <radialGradient
      id="fff000"
    >
      <stop
        offset="0%"
        stopColor="#fff"
      />
      <stop
        offset="100%"
        stopColor="#000"
      />
    </radialGradient>
  </defs>
  <circle
    cx={100}
    cy={100}
    fill="url(#fff000)"
    r={96}
    stroke="#fff"
    strokeWidth="2"
  />
</svg>
`;

exports[`<TokenImage /> renders and matches the snapshot 1`] = `
<svg
  height={100}
  version="1.1"
  width={100}
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <radialGradient
      id="fff000"
    >
      <stop
        offset="0%"
        stopColor="#fff"
      />
      <stop
        offset="100%"
        stopColor="#000"
      />
    </radialGradient>
  </defs>
  <circle
    cx={50}
    cy={50}
    fill="url(#fff000)"
    r={46}
    stroke="#fff"
    strokeWidth="2"
  />
</svg>
`;
